library/std/src/sys/net/connection/socket/unix.rs

```diff
@@ -362,11 +406,52 @@ impl Socket {
     }
 
     #[cfg(any(target_os = "android", target_os = "linux", target_os = "cygwin"))]
-    pub fn recv_msg(&self, msg: &mut libc::msghdr) -> io::Result<usize> {
+    pub fn recv_msg_(&self, msg: &mut libc::msghdr) -> io::Result<usize> {
         let n = cvt(unsafe { libc::recvmsg(self.as_raw_fd(), msg, libc::MSG_CMSG_CLOEXEC) })?;
         Ok(n as usize)
     }
 
+    // user is responsible of aligning cmsg_buf to cmsghdr's alignment (use align_of)
```

Comment:
this should be `unsafe` if it depends on the alignment of `cmsg_buf` being more than 1.

Reply:
Should be good now